guides/source/active_record_encryption.md

```diff
@@ -51,71 +59,174 @@ config.active_record.encryption.key_derivation_salt = ENV["ACTIVE_RECORD_ENCRYPT
 
 ```ruby
 class Article < ApplicationRecord
   encrypts :title
 end
-````
+```
 
-The library will transparently encrypt these attributes before saving them in the database and will decrypt them upon retrieval:
+Active Record Encryption library will transparently encrypt these attributes before saving them to the database and will decrypt them upon retrieval. For example,
 
 ```ruby
 article = Article.create title: "Encrypt it all!"
 article.title # => "Encrypt it all!"
 ```
 
-But, under the hood, the executed SQL looks like this:
+However, in the Rails console, the executed SQL looks like this:
 
 ```sql
 INSERT INTO `articles` (`title`) VALUES ('{\"p\":\"n7J0/ol+a7DRMeaE\",\"h\":{\"iv\":\"DXZMDWUKfp3bg/Yu\",\"at\":\"X1/YjMHbHD4talgF9dt61A==\"}}')
 ```
 
-#### Important: About Storage and Column Size
+### Querying Encrypted Data: Deterministic vs. Non-deterministic Encryption
+
+By default, Active Record Encryption is non-deterministic, which means that encrypting the same value with the same key twice will result in *different* encrypted values (aka ciphertexts). The non-deterministic approach improves security by making crypto-analysis of ciphertexts harder.  However, it makes querying the database impossible.
+
+If you need to able to query the encrypted `email` field on the `Author` model below, you can use deterministic encryption:
+
+```ruby
+class Author < ApplicationRecord
+  encrypts :email, deterministic: true
+end
 
-Encryption requires extra space because of Base64 encoding and the metadata stored along with the encrypted payloads. When using the built-in envelope encryption key provider, you can estimate the worst-case overhead at around 255 bytes. This overhead is negligible at larger sizes. Not only because it gets diluted but because the library uses compression by default, which can offer up to 30% storage savings over the unencrypted version for larger payloads.
+# You can only query by email if using deterministic encryption.
+Author.find_by_email("tolkien@email.com")
+```
 
-There is an important concern about string column sizes: in modern databases the column size determines the *number of characters* it can allocate, not the number of bytes. For example, with UTF-8, each character can take up to four bytes, so, potentially, a column in a database using UTF-8 can store up to four times its size in terms of *number of bytes*. Now, encrypted payloads are binary strings serialized as Base64, so they can be stored in regular `string` columns. Because they are a sequence of ASCII bytes, an encrypted column can take up to four times its clear version size. So, even if the bytes stored in the database are the same, the column must be four times bigger.
+The `deterministic:` option generates initialization vectors in a deterministic
```

Comment:
I guess you describe deterministic here, I wonder if it could be explained earlier on, at the first mention